backend/app/store/image/image.go

```diff
@@ -164,22 +164,28 @@ func (s *Service) ExtractPictures(commentHTML string) (ids []string, err error)
 				}
 			}
 			if strings.Contains(im, s.ProxyAPI) {
-				proxiedURL, err := url.Parse(im)
-				if err != nil {
+				proxiedURL, e := url.Parse(im)
+				if e != nil {
+					err = errors.Wrapf(e, "problem parsing %s", im)
 					return
 				}
-				imgURL, err := base64.URLEncoding.DecodeString(proxiedURL.Query().Get("src"))
-				if err != nil {
+				imgURL, e := base64.URLEncoding.DecodeString(proxiedURL.Query().Get("src"))
+				if e != nil {
+					err = errors.Wrapf(e, "problem decodind base64 from %s", im)
 					return
 				}
-				imgID, err := CachedImgID(string(imgURL))
-				if err != nil {
+				imgID, e := CachedImgID(string(imgURL))
+				if e != nil {
+					err = errors.Wrapf(e, "problem retrieving cached URL for %s", im)
 					return
 				}
 				ids = append(ids, imgID)
 			}
 		}
 	})
+	if err != nil {
```

Comment:
I'm not sure if the error check here safe/correct. I mean, extraction before this change worked even for partially valid document and returned some images anyway. I think it was intentional, and now we may have all or nothing

Reply:
Now it's all or nothing, but there is no way you get an error on `submitImages` (the only function using that function before the PR) after comment sending now because post sending calls the same function and won't allow posting incorrect images.

I don't recall an intention of ignoring the error when I wrote cache image extraction, when I was preparing that PR I was surprised by the bare return inside that function returning null instead of `err`

Reply:
hmm, the function ExtractPictures returned an error only in one case - goquery.NewDocumentFromReader failed. Everything else was forgiving and func tried to extract as many pictures as possible from the comment. After this change instead of valid pictures, submitImages won't get anything if one of the images failed, and idsFn will return an empty slice. Are you sure this behavior valid?

Is this change even relevant to the PR? I don't see how changing the logic of ExtractPictures is necessary for the proper validation.

Reply:
Without that change, we do not validate cached images. But they are available in the comment only on edit and not on the post and we don't validate edits in this PR.

I think you are right and it would be better to revert that function change and document it better instead.

Reply:
Now, as we talk about it, I started wondering if we should return the error at all: it's confusing to have an error of HTML parsing returned to the caller but cached image parsing errors hidden in the code. Making it errorless will clarify the intention of returning ids at any cost, what do you think?

Reply:
the error it returns now is kind of "panic case, should never happen". Logging this thing as we do is ok strategy, just in case.